### ISSUE TITLE
Linux: Crash when creating window if more than 1024 file descriptors are open

### ISSUE THREAD

**Issue by fvallee-bnx** [SFML/SFML#1900]
## Subject of the issue

My (linux) application using SFML is opening many files (not using SFML at all) then create a SFML Window. Everything works fine if I open less than FD_SETSIZE (1024)  files.
If I open more than 1024 files (I'm using ulimit -n 4096) the application crashes in SFML when creating the window.

## Your environment

* Linux (Ubuntu 20.04)
* SFML 2.5.1 (ubuntu package or compiled from 2.5.1 tag)
* gcc 9.3
* ulimit -n 4096

## Steps to reproduce

Using SFML Release build, any app opening files then creating a window:

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    /* do something to open more than 1024 files so file descriptor values are > 1024 */
....

    sf::RenderWindow window(sf::VideoMode(1280, 720), "Minimal, complete and verifiable example");
    window.setFramerateLimit(60);

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        window.display();
    }
}
```

## Expected behavior

No crash and SFML window created

## Actual behavior

Crash with "*** buffer overflow detected ***: terminated" message.

Crash comes from hasMonitorEvent() in src/SFML/Window/Unix/JoystickImpl.cpp 

```cpp
     int monitorFd = udev_monitor_get_fd(udevMonitor);
     fd_set descriptorSet;
     FD_ZERO(&descriptorSet);
     FD_SET(monitorFd, &descriptorSet);
     timeval timeout = {0, 0};
      return (select(monitorFd + 1, &descriptorSet, NULL, NULL, &timeout) > 0) &&
               FD_ISSET(monitorFd, &descriptorSet);
```
In my case monitorFd is > FD_SETSIZE and FD_SET crashes (FD_SET / select crash when the file descriptor is > FD_SET_SIZE  is a well know linux issue)

For some reason I can't explain I don't reproduce the crash with a debug SFML build (i.e. CMAKE_BUILD_TYPE=Debug) even if the monitorFd has the exact same value.

**Comment by fvallee-bnx** [comment]
FYI as I don't care about joysticks  I've added a simple 

`   
 if (monitorFd > FD_SETSIZE)
       return false;
`

check to workaround the issue and it seems good enough

**Comment by eXpl0it3r** [comment]
The mentioned code snippet:

https://github.com/SFML/SFML/blob/87e84bc9e55cc33d3a3bbb06032893e6534a61ba/src/SFML/Window/Unix/JoystickImpl.cpp#L251-L263

**Comment by eXpl0it3r** [comment]
Fixed with #1941 on the 2.6.x branch and will later be merged down to master
